pkg/attestlib/e2etest/pkix_e2e_test.go

```diff
@@ -0,0 +1,144 @@
+fPaxMyZWZ6ZUyKjyICAhcOF90mYAkkJQbrW8ftkU4k1bBU+JfYcTiILyq4tPdrZb
+mHzU6INjYVd8Pm9kHSXJxc1mhqzKFSULhMWwvTW2CHtbtY3mIEOXDFjXdtyC/v1I
+huujP8L6RLX5yq7gZBeYQctgWDzSlBV7uywFpDerEDQcvSF5cl8zRh7jQDDdoLjf
+SH3Z/ZrkmBjwkSpFO1iCHdVW6rObcvhC9JaIfN8Hb0FcIHQay0UrjjxpiSTZE8jp
+DUuShPs2IKq49UXURidRcqsL4A2hav+XuV0QntXi+mj70QKWddzEDOr+Sep43lZ0
+zqI9ifKrAgMBAAECggEAX3wmuYjxNPKSAFfNwbl26y0w0LmOg+/E8bKn+bx1UsQj
+UgR66oaLekFfBeH2mz7HegKbOSl2DHbBX3V8SX5Xb99dXIpFKLWy9//D0qNsKnYU
+QneGU3gb1gEt1PoJLHo8RhqMmMLmJushSZi5VjNvfLvMBqSkaGHj/Y1VDSXw8v3q
+w25PxoudycauNLW0CKLEd2QuPHyjd/krc4R3364PRuIS524V8f+TG3zbXYP+0el3
+oPxApXHuMcbwSN+SqyHJecYt5LBVjhJNjilG1p9V4FMzeMnDeYLbi6PLkObtCye5
+G8yT2pMIgYztCcbOVrVDPnQ3RqJXuz5r4KHLn9c+QQKBgQD02oHfjSWlLBIOMohh
+A1nb3ffsskF6g8aIKz233sGvCrHDyIACzvgwAhV//crp5xuRftyMk0bDpLKFwcvD
+Z0BXKgPntq1IRzDyZyKggXEp/XjtPFs6HHFovyX7a1N3/1J20hJJ+gEVEDXAhqxv
+88cB+92vFQVbVxotRpuqwEtNaQKBgQDfDkaAeuiX00NXWoymgJRU4hg1vxdcEQ7X
+L7JHtUqF510MWsiZBpHfRAV7EN1ZlvGxdBRk2bbTZwV3+4X89SLJLdhS/3fkix1Z
++h/UFm9HZZxcb2UMj/FTuRdQhj1a7h/l6v4KQ2qVCkxwH4Ih5UvLxdmnIcYpbFoB
+UhuKGYy48wKBgBucIBTzN5F+fLK5JAO8ev998hzwpM2J2nM0XQtExRZV4GGxVC38
+5KOVOJNsLNtfQg6P/ZKkcNBS3AaVKcLo+6pYhIQ2ZyfFT1GmK5NpxTb07BPkQG03
+E9q54zCMk6buVYAsg/9vf3u+jjEdfohJNuGUYxUBC8+W7M66LYd9k4AxAoGAemkd
+K9lH2DpqlH/u/FlIDiVeX/CU40PBPzq/vjaxxTapi+QufjxqxXpQ/67LqyT5Uwd/
+QbFxcibxi10PtTjadEmilDn7FAN8giqQWRZOz4HmA5xmejRsniPPtbNV7JcODmGe
+NQe2cECMnmPMSMnLhPL5dY1FcOozotHqpylKXx0CgYBpaBAZeihs5lCZLG9oyeFD
+ujesvS5UdBomDgcDmmhd8dtZOIKsgGuEgbcG8NROrG0tE38JgJ2UxL9xVTYrEuy7
+TCepA1Coedo2Kx1dnBTyvKBfI/TAsLt+JZ38zymUbvrflMG9c9JlhAtNtxm4MF0B
+oujjRhvcxXGmqIt4OSjdQw==
+-----END PRIVATE KEY-----`
+
+const rsa2048PubKey = `-----BEGIN PUBLIC KEY-----
+MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA1Vf+g3iALHB5GHz2sTMm
+VmemVMio8iAgIXDhfdJmAJJCUG61vH7ZFOJNWwVPiX2HE4iC8quLT3a2W5h81OiD
+Y2FXfD5vZB0lycXNZoasyhUlC4TFsL01tgh7W7WN5iBDlwxY13bcgv79SIbroz/C
++kS1+cqu4GQXmEHLYFg80pQVe7ssBaQ3qxA0HL0heXJfM0Ye40Aw3aC430h92f2a
+5JgY8JEqRTtYgh3VVuqzm3L4QvSWiHzfB29BXCB0GstFK448aYkk2RPI6Q1LkoT7
+NiCquPVF1EYnUXKrC+ANoWr/l7ldEJ7V4vpo+9EClnXcxAzq/knqeN5WdM6iPYny
+qwIDAQAB
+-----END PUBLIC KEY-----`
+
+const rsa3072PubKey = `-----BEGIN PUBLIC KEY-----
+MIIBojANBgkqhkiG9w0BAQEFAAOCAY8AMIIBigKCAYEA174cl1EgPn1w/4r1OTNy
+O5ViVEtjPV1Bl2xXA/4EOcVyvztdwovJriC2sry+gW7WS/YraS6BXTY328daY7u5
+OYAHnr/NiF3BoAzbAYMpHU6otMiAlGADrgevESj5XKUp4+XuiYxdOPH7pqgHaZ0+
+ZIvp08vH+xXsY1+WqA3TBsKxkCwTmWGUSy/j2Gml6XHgX1SywJGJ9VEI/5wACExX
+3NYHx+OXBS5XyjFzIoBxFiuA4g8FRHcTw1uulX98Rt/WTwTjesMeWaB7mOxgK03A
+rTOil11+/Wgboow3V93B5pdNMvVjkahYHvkjYkj/HJRuXUKidZANhBWEewpdHDkK
++NXa8BlxcbkzcO2DvXGRu1g0emVInoIEN+/1fz4ab/mTWUGgs9QX+INChlSV3EBz
+5I6mXbc3fE4EW2W9KASIPKIpF1HBaF58B//W+XAy7mxhuSfg23/mkf4SI9ZVM82s
+rBmJaTmRgKWbZXNaf4yH+RhyGF1CsBN2uLrt667Swxv3AgMBAAE=
+-----END PUBLIC KEY-----`
+
+func TestPkixEndToEnd(t *testing.T) {
+	tcs := []struct {
+		name                string
+		privateKey          []byte
+		publicKey           []byte
+		signatureAlgorithm  attestlib.SignatureAlgorithm
```

Comment:
all tests share same value for this as well.

for coverage purpose, we could possibly test if the verifier and signer uses different `signatureAlgorithm`, e.g., with different sha values, verification should fail.

Reply:
done. This case is covered in a unit test where the attestation algorithm differs from the one declared on the public key.

Reply:
Actually, I went back and checked this is not covered in the unit tests. I think we should add it in the unit rather than here as that test the functionality of the verifier.